core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -144,25 +160,30 @@ public JoinDefinition define(AutoJoinerContext context) {
   public void prepareRun(BatchJoinerContext context) {
     if (conf.getNumPartitions() != null) {
       context.setNumPartitions(conf.getNumPartitions());
+      if (conf.getDistributionFactor() != null && conf.getDistributionFactor() < conf.getNumPartitions()) {
+        LOG.warn(String.format(
+          "Number of partitions (%d) should be greater than or equal to distribution factor (currently %d) "
```

Comment:
Use please holder `{}` instead of `String.format` when using Logger.

Reply:
done